### DISCUSSION TITLE
extend not working with PostCSS and Webpack

### DISCUSSION THREAD

**Discussion by dskvr** [tailwindlabs/tailwindcss#10419] [Help]
**What version of Tailwind CSS are you using?**

Latest

**What build tool (or framework if it abstracts the build tool) are you using?**

`postcss-apply:0.12.0`
`webpack:5.75.0` 

**What version of Node.js are you using?**

latest (19)

**What browser are you using?**

All

**What operating system are you using?**

All

**Reproduction URL**

Cannot because tailwind play does not provide the capability to reproduce the problem 

**Describe your issue**

#2721 Should not have been closed, this issue can be reproduced.

Using postcss with webpack, the `extend` options **are completely ignored**. When `postcss.config.js` is deleted, _all `@apply` statements are removed_ and removed from webpack config ofc, everything works as expected. 

I have attempted:
1. PurgeCSS for PostCSS - made things worse actually
2. Rummaged through PostCSS docs to try and figure out what is going on. 

There is no documentation on this case that I can find, there are no caveats listed [here](https://tailwindcss.com/docs/customizing-spacing), and it is a pretty common case to want to use postcss, who doesn't want to use `@apply`? 

**Hack:**
For anyone that needs all values between 1-32 for margin-top and/or height and has already wasted enough time: 
https://gist.github.com/dskvr/e4d34bc5c7e96fd34f62d947aa3939b7

**Edited** Forgot to mention I have to remove all `@apply` statements as well, if I don't, the project won't build.

**Comment by adamwathan** [comment]
Hey! Please provide a reproduction, just create a small project that demonstrates the problem and push it up to a repository on GitHub.

You don't need `postcss-apply` by the way, that's a totally different `@apply` than the one that is built-in to Tailwind.

**Reply by dskvr** [reply]
Interesting, if I don't use `postcss-apply` I cannot build, once introducing `postcss-apply` `@apply` works as expected. Not sure what I'm missing. I'm in the middle of feature development rn, so cannot provide replication at this moment in time,. That said, I definitely want to get to the bottom of this, especially if I can remove `postcss-apply` and reap the benefits of a smaller, albeit slightly, bundle.  Will return in due time.

**Comment by adamwathan** [comment]
Here's an example project with Tailwind CSS + webpack 5 that uses `extend` and generates the CSS as expected. Clone it, run `npm install`, then `npm run build`, then inspect the generated CSS in `./dist/main.css` to see that the customized values are included.

https://github.com/adamwathan/issue-10419
